### ISSUE TITLE
fix how tree command copies data to the output buffer

### ISSUE THREAD

**Issue by trgill** [sid-project/sid#97]
```
	buffer_get_data(readbuf, (const void **) &msg, &size);
		if (size < USID_MSG_HEADER_SIZE || msg->status & COMMAND_STATUS_FAILURE) {
			buffer_destroy(readbuf);
			return -1;
		}
		size -= USID_MSG_HEADER_SIZE;
		buffer_add(outbuf, msg->data, size, &r);
```
Comment from @prajnoha : 

Hmm, we're reading almost the whole complete and formatted message here, just skipping the header. Then we're copying all the rest to the output buffer. It works fine, but would be nicer if we could avoid the extra copy.

Maybe we could define the output buffer as BUFFER_TYPE_VECTOR in which case we could add the pointer to the msg->data + size directly into the vector buffer as element. But right now, formatting is done on sidctl side for other command now and they use the output buffer for formatting so there the BUFFER_TYPE_VECTOR is not ideal. But this is caused by the fact that tree command is formatted on SID side while the rest is formatted on sidctl side at the moment.

**Comment by bmarzins** [comment]
As part of my library API work, I'm having SID output formatted ascii as the reply data from its commands. This will allow us to change sidctl to simply print the output from the server directly, without needing to process it into a different buffer.

**Comment by prajnoha** [comment]
Yup, commands/clients staying thin then and just asking for the exact format that SID will prepare for them so they don't need to bother - the formatting will get more 'centralized', so easier to manage. Then the "response format request" becomes part of the API/protocol between cmds/clients and SID.

**Comment by prajnoha** [comment]
Fixed with: f684989
